Fix timeshift issue

ak476519 opened the pull request:
Documented time shift issue fixed #476. 

The issue was caused by timestamps without timezones and a localization feature of moment.js.
After evaluating different solutions, a easier fix was selected because of the following reasons:
1. The package that is causing the problem, moment.js, has already stopped updating and its functionality in this project would be replaced with other packages such as JavaScript's updated date library in the following years. It's more feasible to use this time in other goals that are more relevant to long term.
2. This easier (and the cleaner) fix, has minimal changes to the codebase. This would make it easier to convert to other packages in the future. 

Note merging this changes the node modules.

ak476519 committed:
add straight-forward fix to timeshift:
function changeTimezoneOffset
will make line graph act normal

ak476519 committed:
remove print commands used for testing purposes,
add comments

huss commented on src/client/app/containers/LineChartContainer.ts:
```diff
@@ -30,8 +31,9 @@ function mapStateToProps(state: State) {
 				const hoverText: string[] = [];
 				const readings = _.values(readingsData.readings);
 				readings.forEach(reading => {
+					// const timeReading = recreateTimezoneOffset(reading.startTimestamp, 'America/Chicago').tz('America/Chicago');
```
Should this commented out line and the import be removed?

huss replied:
done.

huss commented on src/client/app/utils/timezoneHandler.ts:
```diff
@@ -0,0 +1,24 @@
+ * Checks if moment-timezone can recognize the provided timezone name
+ * @param timezone
+ */
+function isValidTimezone(timezone: string): boolean {
```
I don't see where this is used.

huss replied:
Added comment but left in case needed for export PR.

huss commented on src/client/app/utils/timezoneHandler.ts:
```diff
@@ -0,0 +1,24 @@
+ * recreate a timestamp created in default timezone(UTC) with its true timezone
+ * this is invoked only when the user wishes to export data in their absolute time in UNIX timestamps
+ */
+export function recreateTimezoneOffset(origin: number, targetZone: string): moment.Moment {
```
This only seems used by commented out code in LineChartContainer.ts (see comment there). Not sure if this is needed in the end.

huss replied:
See other comment on leaving for now.

huss reviewed (changes requested):
For some reason I cannot add any more comments to some parts of the code so I'm leaving them here. I hope I have the descriptions correct and I'm correct about what I saw.

- meters work fine with lines. With bar the values are all fine but the dates may be shifting as described for groups below.
- groups with lines have the correct value but the time is earlier by the shift from UTC. I think the same change for meters needs to be applied to groups.
- groups with bar chart has the wrong value. My testing indicates that they are actually summing the right number of points but they are all shifted earlier in a similar way to lines. So, you get 24 points for a day bar but it is the wrong range of points.
- groups with bar chart also can have the wrong days. This screenshot:
![Screen Shot 2021-01-19 at 12 06 45 PM](https://user-images.githubusercontent.com/987062/105076821-34d29980-5a51-11eb-839e-625a88b53a5a.png)
shows that the groups bar (it was the sum of the two meters displayed) has different days than the meters. I suspect that the calculation of the days involved has a timezone issue for the groups (and for meters as noted above).
- compare graphs have a timezone issue. The range of values summed seem to shift forward by the UTC timezone difference. This is the opposite direction than groups with bar. This also applies to meters here.
- exporting is working fine.
- There are some linting errors but I think they will go away when the newer development is merged (there will be conflicts).
- testing is passing.

I'm asking for @ak476519 to look at these issues. Once complete I will recheck all values and can then approve this PR. If anything is not clear or you cannot see what I say then just let me know.

huss committed:
Merge branch 'development' of https://github.com/OpenEnergyDashboard/OED into pr/ak476519/508

huss committed:
address PR comment on unused code

huss committed:
fix group line date value

huss committed:
fix group bar times

This fixes the issue in the comment in the PR that groups have shifted
times. Bar values for meters are fine (maybe always were and other issue
now solved). Same is true for bar with wrong days as
they are now fine.

huss committed:
fix compare times

- This now takes care of the moment shift between UTC and local time for
 compare graphics.
 - Removed conn that was not used and incorrect passing to a function.
 - fixes seeming copy mistake where FourWeeks set to 1 day rather
than 28 days.

huss committed:
fix dates on map

Makes dates correct for the local time shift of moment.
Basically the same as bar graphic since uses that data.

huss commented:
I've merged this with development. I addressed the issues stated in the PR. I also fixed for similar issue with maps. Local check/testing is fine. This should be ready to merge.
